src/Otp.php

```diff
@@ -91,9 +91,20 @@ public function totp($secret, $timecounter = null)
     /* (non-PHPdoc)
      * @see Otp.OtpInterface::checkHotp()
     */
-    public function checkHotp($secret, $counter, $key)
+    public function checkHotp($secret, $counter, $key, $counterwindow = 5)
```

Comment:
I think this should be it's own function, leaving this one unchanged. Now everyone suddenly does this check for the counter window and that's bad. Also, the window should be 2 or 3 at most for a default, like the RFC suggests as well.
Not a fan of dual returns either very much, but in this case I think false for failure would make sense, though the return for the other case is not 100% clear. Think of it this way: You enter code for counter one, but it succeeds on counter 2, the function returns 2, then the system blocks the code for 1 temporary to prevent replay attack, but code 2 could still be used right away. That's also something that is technically bad with the time based one, but there your counter changes automatically and you rarely need to login in twice within that 30 seconds frame. With counter based, the time frame is not given and so it;s a bit more dangerous.